pipit/tests/trace.py

```diff
@@ -97,5 +103,9 @@ def test_match_caller_callee(data_dir, ping_pong_otf2_trace):
 
     df = trace.events
 
-    # all events of the ping pong trace are roots with no children
-    assert len(df.loc[(df["Event Type"] == "Enter") & (df["_parent"].notnull())]) == 0
+    # 40 events across both ranks that occur within main
+    assert len(df.loc[df["_parent"].notnull()]) == 40
+
+    # 2 main functions that are also roots and only events
+    # that have children in the trace
+    assert len(df.loc[df["_children"].notnull()]) == 2
```

Comment:
we can make this part simpler:
nodes with no parents =2
nodes with no children = 40